SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -1579,16 +1578,24 @@ void USpatialNetDriver::ProcessRPC(AActor* Actor, UObject* SubObject, UFunction*
 		return;
 	}
 
-	FUnrealObjectRef CallingObjectRef = PackageMap->GetUnrealObjectRefFromObject(CallingObject);
+	const FUnrealObjectRef CallingObjectRef = PackageMap->GetUnrealObjectRefFromObject(CallingObject);
 	if (!CallingObjectRef.IsValid())
 	{
 		UE_LOG(LogSpatialOSNetDriver, Warning, TEXT("The target object %s is unresolved; RPC %s will be dropped."),
 			   *CallingObject->GetFullName(), *Function->GetName());
 		return;
 	}
-	RPCPayload Payload = Sender->CreateRPCPayloadFromParams(CallingObject, CallingObjectRef, Function, Parameters);
 
-	Sender->ProcessOrQueueOutgoingRPC(CallingObjectRef, MoveTemp(Payload));
```

Comment:
What was the reasoning behind pulling the cross server RPC out of the sender? When we implement Cross Server RPC based on state, we might want them to follow the same path as the other RPCs (to get the queuing behaviour on overflow for instance).

Reply:
(what might happen is that the sender and handler you wrote get moved to the RPC service that Jacques has re-written.)

Reply:
yeah, and that is ok. At the moment cross server RPCs are not based on state and have a fairly different flow compared to all other RPCs, so it doesn't make much sense to have it in the RPC Service.
The reason it got pulled out of the Sender and the receiver is because the overall flow at the moment is hard to follow and reason about as there is just so much branching going on (e.g. it took me a while to identify that we actually never queue cross server RPCs).

So the sender and handler are mainly here to simplify the overall behaviour. when we then implement them based on state, we probably can get rid of this.

Reply:
It could be worth moving creating payload logic (which seems to be shared between both cross server and not) into some utilities class of the RPC service, then move the cross server RPCs into the RPC service as well. Right now in the rewrite though, sending is not fully handled by the RPC service, so it probably doesn't make sense yet. 

For reference, I was thinking the RPC service would be more of a logical encapsulation of all things RPCs rather than unifying flows, so that there's one central place where a developer can go to reason about them when making changes/debugging. The multicast and client server ones are more similar to each other than to cross server RPCs, but still fairly different.